packages/runtime/src/routeGeneration/templateHelpers.ts

```diff
@@ -618,41 +618,55 @@ export class ValidationService {
   }
 
   /**
-   * combine all schemas once without backwards combinations ie
+   * combine all schemas once, ignoring order ie
    * input: [[value1], [value2]] should be [[value1, value2]]
    * not [[value1, value2],[value2, value1]]
    * and
-   * input: [[value1], [value2], [value3]] should be [
-   *   [value1, value2, value3],
-   *   [value1, value2],
-   *   [value1, value3],
-   *   [value2, value3]
+   * input: [[value1, value2], [value3, value4], [value5, value6]] should be [
+   *   [value1, value3, value5],
+   *   [value1, value3, value6],
+   *   [value1, value4, value5],
+   *   [value1, value4, value6],
+   *   [value2, value3, value5],
+   *   [value2, value3, value6],
+   *   [value2, value4, value5],
+   *   [value2, value4, value6],
    * ]
    * @param modelSchemass
    */
-  private selfIntersectionExcludingCombinations(modelSchemass: TsoaRoute.RefObjectModelSchema[][]): TsoaRoute.RefObjectModelSchema[] {
+  private selfIntersectionCombinations(modelSchemass: TsoaRoute.RefObjectModelSchema[][]): TsoaRoute.RefObjectModelSchema[] {
     const res: TsoaRoute.RefObjectModelSchema[] = [];
-
-    for (let outerIndex = 0; outerIndex < modelSchemass.length; outerIndex++) {
-      let currentCollector = { ...modelSchemass[outerIndex][0] };
-      for (let innerIndex = outerIndex + 1; innerIndex < modelSchemass.length; innerIndex++) {
-        currentCollector = { ...this.intersectRefObjectModelSchemas([currentCollector], modelSchemass[innerIndex])[0] };
```

Comment:
The bug was caused on lines 637 and 639, using the first subschema from each schema of the intersection